### ISSUE TITLE
Adding ringbuf versions of existing libbpf-tools

### ISSUE THREAD

**Issue by krisztianfekete** [iovisor/bcc#4260]
Hi,

Yesterday, at this year's eBPF Summit @eiffel-fl suggested contributing our [ringbuf based ports ](https://github.com/solo-io/bumblebee/tree/main/examples)to this repositoryin some way if it's beneficial to the community.

We started to brainstorm about the different approaches and tradeoffs and we came to this conclusion that best effort/value solution would be to generate the desired (`BPF_MAP_TYPE_RINGBUF` or `BPF_MAP_TYPE_PERF_EVENT_ARRAY`) version with `#ifdef`.

The idea of creating multiple maps in kernelspace, adding another version of the same tools with a suffix like `_ringbuf.bpf.c`, and generating the required map in userspace based on the kernel version were discarded.

What do you all think, would it make sense to add these in some way? If so, what approach would be the best on the long run?

**Comment by chenhengqi** [comment]
Nope, we could achieve this more elegantly. Please refer to the libbpf `bpf_map__set_autocreate` API.

**Comment by chenhengqi** [comment]
Just opened #4262 .

**Comment by krisztianfekete** [comment]
Hey, @chenhengqi I just opened this https://github.com/iovisor/bcc/pull/4272.
